src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -31,25 +31,36 @@ public class TemplateConverter
         private readonly IFileResolver fileResolver;
         private readonly Uri fileUri;
         private readonly JObject template;
+        private readonly Dictionary<ModuleDeclarationSyntax, Uri> jsonTemplateUrisByModule;
 
-        private TemplateConverter(Workspace workspace, IFileResolver fileResolver, Uri fileUri, JObject template)
+        private TemplateConverter(Workspace workspace, IFileResolver fileResolver, Uri fileUri, JObject template, Dictionary<ModuleDeclarationSyntax, Uri> jsonTemplateUrisByModule)
```

Comment:
Obviously a bit of an edge case, but do you know what would happen if you have linked templates with same path & filename but different extension? Would the generated Bicep files overwrite each other?

Reply:
Good catch! Yes they will overwrite each other in that case. I've add code to handle it by appending `.bicep` to the JSON path if collision is detected (tests: https://github.com/Azure/bicep/pull/2201/files#diff-c0a7fdc6ac0b15f94bc349d62098ae6d83f1a816c9db18f138ea6dd5a6af2d63).

Reply:
Hmm...some weird race condition is going on with the tests. I'll need to revisit the fix.

Reply:
Turns out `GetWorkingExampleData` also returns nested templates, so they will get decompiled twice in the tests. I've update the method to filter out nested templates, and the tests are passing consistently now.